indigo_api/migrations/0129_migrate_refs_uris.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2020-04-22 12:39
+from __future__ import unicode_literals
+
+from lxml import etree
+
+from django.db import migrations
+
+from cobalt import Act, FrbrUri, datestring
+from cobalt.uri import FRBR_URI_RE
+from indigo.analysis.refs.base import BaseRefsFinder
+
+
+def new_frbr_uri(uri, forward):
+    """ Sets prefix on uri:
+        'akn' if forward is True, None if it's False.
+    """
+    if not isinstance(uri, FrbrUri):
+        uri = FrbrUri.parse(uri)
+    uri.prefix = 'akn' if forward else None
+    return uri
+
+
+class RefsFinderRef(BaseRefsFinder):
```

Comment:
I'm not sure that BaseRefsFinder is doing anything useful for you here. If your xpath is restricted to `//a:ref`, then `ref` tags can only occur in textual elements, so you don't need any of the ancestor stuff. At that point, `handle_refs` doesn't need anything from the parent class and it can be a simple function.

Reply:
If I restrict the xpath to `//a:ref`, the hrefs in e.g. `passiveRef` in `meta` would need to be updated separately, right?

Reply:
But if I were to keep the existing `candidate_xpath`, are you saying I could just do something like,
for node in root where there's an href starting with '/', do the right thing?

Reply:
> If I restrict the xpath to `//a:ref`, the hrefs in e.g. `passiveRef` in `meta` would need to be updated separately, right?

I had forgotten about those. As long as you're sure that all `href` attributes are safe to process as AKN URIs, then we could do them all in one go. 

> But if I were to keep the existing candidate_xpath, are you saying I could just do something like, for node in root where there's an href starting with '/', do the right thing?

Yes. Although I don't think you need the candidate xpath at all. Just process every `ref` (or element with an `href`) in the entire document.

Reply:
so just checking, I can be namespace agnostic here? 
```
    root = etree.fromstring(document.document_xml)
    # path = etree.XPath("//a:*[starts-with(@href, '/')]", namespaces={'a': cobalt_doc.namespace}) <-- don't need this?
    path = etree.XPath("//*[starts-with(@href, '/')]")
    for node in path(root):
```

Reply:
Sorry, no, I meant all AKN elements with a an `href` attribute. I can't think of a time when we'd want to ignore namespaces.